### ISSUE TITLE
Bayesian VAR

### ISSUE THREAD

**Issue by clgarciga** [statsmodels/statsmodels#7449]
If I wrote a basic Bayesian VAR module, could I submit it for inclusion into statsmodels?

**Comment by ChadFulton** [comment]
I think this would be a good addition, especially if the initial implementation focused on the natural conjugate priors and 
closed form posteriors. Then it could be like the Bayes Mixed GLM code (https://github.com/statsmodels/statsmodels/blob/main/statsmodels/genmod/bayes_mixed_glm.py)

We have so-far stayed away from sampling (leaving that to e.g. pymc), and while I don't expect that to always be the case (particularly since Gibbs sampling is not a priority of pymc), we need to think more carefully about how to handle that.

I think you could start by creating a pull request that implements the basic econometrics, and we could iterate from there (but see below for where we would likely need to end up for the PR to be merged).

---------

Just to prepare you, there are a couple of things to note that sometimes trip up newcomers and stall pull requests:

- We follow the Python style guidelines (e.g. pep 8), so new code will have to be conforming
- Code must be unit tested, almost always either using analytic results or results from some other established package
- Ultimately, the new functions / classes should broadly fit into the existing Statsmodels code style (e.g. model classes and results objects, etc.)

The second two things often take up a substantial amount of time (sometimes even more time than the implementation of the actual econometrics), and this can be frustrating.

**Comment by ChadFulton** [comment]
And thanks for contributing!

**Comment by clgarciga** [comment]
```
I think this would be a good addition, especially if the initial implementation focused on the natural conjugate priors and
closed form posteriors. 
```

Yup, that's exactly what I was thinking :)

**Comment by ChadFulton** [comment]
Also, I just want to note that you do not need to take care of all the things I mentioned above before starting a pull request. Once the basic functionality is there, we'd be happy to help figure out how to integrate it into Statsmodels, etc.
